Signal/src/view controllers/ExperienceUpgradeViewController.swift

```diff
@@ -0,0 +1,292 @@
+        carouselView.autoPinEdge(toSuperviewEdge: .top)
+        carouselView.autoPinEdge(toSuperviewEdge: .left)
+        carouselView.autoPinEdge(toSuperviewEdge: .right)
+
+        // Build slides for carousel
+        var previousSlideView: UIView?
+        for experienceUpgrade in experienceUpgrades {
+            let slideView = buildSlideView(header: experienceUpgrade.title, body: experienceUpgrade.body, image: experienceUpgrade.image)
+            carouselView.addSubview(slideView)
+
+            slideView.autoPinEdge(toSuperviewEdge: .top, withInset: ScaleFromIPhone5(10))
+            slideView.autoPinEdge(toSuperviewEdge: .bottom)
+            slideView.autoMatch(.width, to: .width, of: carouselView)
+
+            // pin first slide to the superview
+            if previousSlideView == nil {
+               slideView.autoPinEdge(toSuperviewEdge: .left)
+            } else {
+                // pin any subsequent slide to the preveding slide
+               slideView.autoPinEdge(.left, to: .right, of: previousSlideView!)
+            }
+
+            previousSlideView = slideView
+        }
+        // we should never be presenting a blank slideshow.
+        // but if so, we don't want to crash in prod.
+        assert(previousSlideView != nil)
+
+        // ping the last slide to the superview right.
+        previousSlideView?.autoPinEdge(toSuperviewEdge: .right)
+
+        // Previous button
+
+        // Lightening the arrows' color to balance their heavy stroke with the thinner text on the page.
+        let arrowButtonColor = UIColor(red: 0.1, green: 0.1, blue: 0.1, alpha: 1)
+        let previousButton = UIButton()
+        self.previousButton = previousButton
+        splashView.addSubview(previousButton)
+        previousButton.isUserInteractionEnabled = true
+        previousButton.setTitleColor(arrowButtonColor, for: .normal)
+        previousButton.accessibilityLabel = NSLocalizedString("UPGRADE_CAROUSEL_PREVIOUS_BUTTON", comment: "accessibility label for arrow in slideshow")
+        previousButton.setTitle("‹", for: .normal)
+        previousButton.titleLabel?.font = UIFont.ows_mediumFont(withSize: ScaleFromIPhone5To7Plus(24, 48))
+        previousButton.addTarget(self, action:#selector(didTapPreviousButton), for: .touchUpInside)
+
+        // Previous button layout
+        previousButton.autoPinEdge(toSuperviewEdge: .left)
+        let arrowButtonInset = ScaleFromIPhone5(200)
+        previousButton.autoPinEdge(toSuperviewEdge: .top, withInset: arrowButtonInset)
+
+        // Next button
+        let nextButton = UIButton()
+        self.nextButton = nextButton
+        splashView.addSubview(nextButton)
+        nextButton.isUserInteractionEnabled = true
+        nextButton.setTitleColor(arrowButtonColor, for: .normal)
+        nextButton.accessibilityLabel = NSLocalizedString("UPGRADE_CAROUSEL_NEXT_BUTTON", comment: "accessibility label for arrow in slideshow")
+        nextButton.setTitle("›", for: .normal)
```

Comment:
Just FYI - Another option is to use the iOS < "back/prev" and > "next" png assets.  I've already extracted and added the former as "NavBarBack".  NBD, but I think they'd read as more idiomatic.

I used some fork of https://github.com/futuretap/iOS-Artwork-Extractor to do the extraction.